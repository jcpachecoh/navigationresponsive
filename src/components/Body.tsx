import * as React from 'react';

export interface BodyProps {
  mainStyle: React.CSSProperties;
}

export class Body extends React.Component<BodyProps, {}> {
  constructor(props: BodyProps) {
    super(props);
  }

  public render() {
    return (
      <section id="main" className="main">
        <div className="bg">
          <div className="container">
            <div className="banner">
              <h1>Get paid for giving a shit.</h1>
            </div>
          </div>
        </div>
        <div className="text-container">
          <div className="container">
            <h1>Get paid for giving a shit.</h1>
            <p className="text">
              Aute Lorem commodo ipsum dolore laborum enim laborum commodo et
              ea. Aliquip elit velit aliquip in ex laboris consectetur sit
              proident esse cillum. Minim duis ad Lorem excepteur consectetur
              commodo fugiat. Enim consectetur duis ut culpa voluptate labore
              anim eu dolor deserunt non veniam est. Aute fugiat ea nisi eiusmod
              anim ex est et exercitation dolore sint in. Labore magna ullamco
              do dolor sint consectetur. Cillum qui ex do dolor irure magna sint
              ea dolor nulla ad proident aliqua nostrud.
            </p>
          </div>
        </div>
      </section>
    );
  }
}
