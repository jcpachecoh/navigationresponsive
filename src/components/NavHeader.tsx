import * as React from 'react';

export interface NavHeaderProps {
  styleMySideBar: React.CSSProperties;
  styleHeaderMobile: React.CSSProperties;
  closeNav: Function;
  openNav: Function;
}

export interface NavHeaderState {

}

export class NavHeader extends React.Component<NavHeaderProps, NavHeaderState> {
  constructor(props: NavHeaderProps) {
    super(props);
  }

  public render() {
    return (
      <div>
        <div
          className="header-mobile"
          id="header-mobile"
          style={this.props.styleHeaderMobile}
        >
          <div className="logo">HUGE</div>

          <a
            href="javascript:void(0);"
            className="close-icon"
            onClick={() => this.props.closeNav()}
          >
            <i className="fa fa-times" />
          </a>
        </div>
        <div id="mySidebar" className="header">
          <div className="logo">HUGE</div>
          <nav id="nav" className="topnav">
            <ul>
              <li className="icon">
                <a href="javascript:void(0);" onClick={() => this.props.openNav()}>
                  <i className="fa fa-bars" />
                </a>
              </li>
              <li>
                <a href="#home" className="nav-menu">
                  Work
                </a>
              </li>
              <li className="dropdown">
                <a href="#home">About</a>
                <ul className="subnav">
                  <li>
                    <a href="#">What we do</a>
                  </li>
                  <li>
                    <a href="#">How we work</a>
                  </li>
                  <li>
                    <a href="#">Leadership</a>
                  </li>
                </ul>
              </li>
              <li className="dropdown">
                <a href="#home">Carrers</a>
                <ul className="subnav">
                  <li>
                    <a href="#">Client Services</a>
                  </li>
                  <li>
                    <a href="#">Creative</a>
                  </li>
                  <li>
                    <a href="#">Motion & Media</a>
                  </li>
                  <li>
                    <a href="#">Operations</a>
                  </li>
                  <li>
                    <a href="#">People</a>
                  </li>
                  <li>
                    <a href="#">Strategy</a>
                  </li>
                  <li>
                    <a href="#">Technology</a>
                  </li>
                  <li>
                    <a href="#">UX & Product Design</a>
                  </li>
                </ul>
              </li>
              <li className="dropdown">
                <a href="#home">Ideas</a>
                <ul className="subnav">
                  <li>
                    <a href="#">Reports</a>
                  </li>
                  <li>
                    <a href="#">Perspectives</a>
                  </li>
                  <li>
                    <a href="#">Books</a>
                  </li>
                  <li>
                    <a href="#">Videos</a>
                  </li>
                </ul>
              </li>
              <li>
                <a href="#home">News</a>
              </li>
              <li>
                <a href="#home">Events</a>
              </li>
              <li className="dropdown">
                <a href="#home">Contacts</a>
                <ul className="subnav">
                  <li>
                    <a href="#">Atlanta</a>
                  </li>
                  <li>
                    <a href="#">Brooklyn</a>
                  </li>
                  <li>
                    <a href="#">DC</a>
                  </li>
                  <li>
                    <a href="#">London</a>
                  </li>
                  <li>
                    <a href="#">Los Angeles</a>
                  </li>
                  <li>
                    <a href="#">Oakland</a>
                  </li>
                  <li>
                    <a href="#">Toronto</a>
                  </li>
                </ul>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    );
  }
}
