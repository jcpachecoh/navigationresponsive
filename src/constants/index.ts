import { CSSProperties } from '../../node_modules/@types/react';

export const desktopMainStyle: CSSProperties = {
  width: '100%',
  backgroundColor: 'rgba(255, 255, 255)'
};
export const responsiveMainStyle: CSSProperties = {
  width: '25%',
  backgroundColor: 'rgba(0, 0, 0, 0.5)'
};
export const desktopMySideBarStyle: CSSProperties = { width: '100%' };
export const responsiveMySideBarStyle: CSSProperties = { width: '100%' };
export const displayNone: CSSProperties = { display: 'none' };
export const displayBlock: CSSProperties = { display: 'block' };
