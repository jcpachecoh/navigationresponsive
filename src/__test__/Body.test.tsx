import * as React from 'react';
import { shallow } from 'enzyme';
import { Body } from '../components/Body';
import { desktopMainStyle } from '../constants/index';

describe('Body', () => {
    let wrapper: any;
    beforeEach(() => {
        wrapper = shallow(<Body mainStyle={desktopMainStyle} />);
    });
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('contains 1 section tags', () => {
        expect(wrapper.find('section')).toHaveLength(1);
    });
    it('contains  5 div tags', () => {
        expect(wrapper.find('div')).toHaveLength(5);
    });
    it('contains  1 p tags', () => {
        expect(wrapper.find('p')).toHaveLength(1);
    });
    it('contains h1 with text <h1>Get paid for giving a shit.</h1>', () => {
        expect(wrapper.contains(<h1>Get paid for giving a shit.</h1>)).toBe(true);
    });
});
