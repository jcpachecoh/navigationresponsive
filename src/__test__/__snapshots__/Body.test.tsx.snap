// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Body
    mainStyle={
      Object {
        "backgroundColor": "rgba(255, 255, 255)",
        "width": "100%",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="bg"
        >
          <div
            className="container"
          >
            <div
              className="banner"
            >
              <h1>
                Get paid for giving a shit.
              </h1>
            </div>
          </div>
        </div>,
        <div
          className="text-container"
        >
          <div
            className="container"
          >
            <h1>
              Get paid for giving a shit.
            </h1>
            <p
              className="text"
            >
              Aute Lorem commodo ipsum dolore laborum enim laborum commodo et ea. Aliquip elit velit aliquip in ex laboris consectetur sit proident esse cillum. Minim duis ad Lorem excepteur consectetur commodo fugiat. Enim consectetur duis ut culpa voluptate labore anim eu dolor deserunt non veniam est. Aute fugiat ea nisi eiusmod anim ex est et exercitation dolore sint in. Labore magna ullamco do dolor sint consectetur. Cillum qui ex do dolor irure magna sint ea dolor nulla ad proident aliqua nostrud.
            </p>
          </div>
        </div>,
      ],
      "className": "main",
      "id": "main",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <div
              className="banner"
            >
              <h1>
                Get paid for giving a shit.
              </h1>
            </div>
          </div>,
          "className": "bg",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="banner"
            >
              <h1>
                Get paid for giving a shit.
              </h1>
            </div>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1>
                Get paid for giving a shit.
              </h1>,
              "className": "banner",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Get paid for giving a shit.",
              },
              "ref": null,
              "rendered": "Get paid for giving a shit.",
              "type": "h1",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <h1>
              Get paid for giving a shit.
            </h1>
            <p
              className="text"
            >
              Aute Lorem commodo ipsum dolore laborum enim laborum commodo et ea. Aliquip elit velit aliquip in ex laboris consectetur sit proident esse cillum. Minim duis ad Lorem excepteur consectetur commodo fugiat. Enim consectetur duis ut culpa voluptate labore anim eu dolor deserunt non veniam est. Aute fugiat ea nisi eiusmod anim ex est et exercitation dolore sint in. Labore magna ullamco do dolor sint consectetur. Cillum qui ex do dolor irure magna sint ea dolor nulla ad proident aliqua nostrud.
            </p>
          </div>,
          "className": "text-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Get paid for giving a shit.
              </h1>,
              <p
                className="text"
              >
                Aute Lorem commodo ipsum dolore laborum enim laborum commodo et ea. Aliquip elit velit aliquip in ex laboris consectetur sit proident esse cillum. Minim duis ad Lorem excepteur consectetur commodo fugiat. Enim consectetur duis ut culpa voluptate labore anim eu dolor deserunt non veniam est. Aute fugiat ea nisi eiusmod anim ex est et exercitation dolore sint in. Labore magna ullamco do dolor sint consectetur. Cillum qui ex do dolor irure magna sint ea dolor nulla ad proident aliqua nostrud.
              </p>,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Get paid for giving a shit.",
              },
              "ref": null,
              "rendered": "Get paid for giving a shit.",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Aute Lorem commodo ipsum dolore laborum enim laborum commodo et ea. Aliquip elit velit aliquip in ex laboris consectetur sit proident esse cillum. Minim duis ad Lorem excepteur consectetur commodo fugiat. Enim consectetur duis ut culpa voluptate labore anim eu dolor deserunt non veniam est. Aute fugiat ea nisi eiusmod anim ex est et exercitation dolore sint in. Labore magna ullamco do dolor sint consectetur. Cillum qui ex do dolor irure magna sint ea dolor nulla ad proident aliqua nostrud.",
                "className": "text",
              },
              "ref": null,
              "rendered": "Aute Lorem commodo ipsum dolore laborum enim laborum commodo et ea. Aliquip elit velit aliquip in ex laboris consectetur sit proident esse cillum. Minim duis ad Lorem excepteur consectetur commodo fugiat. Enim consectetur duis ut culpa voluptate labore anim eu dolor deserunt non veniam est. Aute fugiat ea nisi eiusmod anim ex est et exercitation dolore sint in. Labore magna ullamco do dolor sint consectetur. Cillum qui ex do dolor irure magna sint ea dolor nulla ad proident aliqua nostrud.",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="bg"
          >
            <div
              className="container"
            >
              <div
                className="banner"
              >
                <h1>
                  Get paid for giving a shit.
                </h1>
              </div>
            </div>
          </div>,
          <div
            className="text-container"
          >
            <div
              className="container"
            >
              <h1>
                Get paid for giving a shit.
              </h1>
              <p
                className="text"
              >
                Aute Lorem commodo ipsum dolore laborum enim laborum commodo et ea. Aliquip elit velit aliquip in ex laboris consectetur sit proident esse cillum. Minim duis ad Lorem excepteur consectetur commodo fugiat. Enim consectetur duis ut culpa voluptate labore anim eu dolor deserunt non veniam est. Aute fugiat ea nisi eiusmod anim ex est et exercitation dolore sint in. Labore magna ullamco do dolor sint consectetur. Cillum qui ex do dolor irure magna sint ea dolor nulla ad proident aliqua nostrud.
              </p>
            </div>
          </div>,
        ],
        "className": "main",
        "id": "main",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
            >
              <div
                className="banner"
              >
                <h1>
                  Get paid for giving a shit.
                </h1>
              </div>
            </div>,
            "className": "bg",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="banner"
              >
                <h1>
                  Get paid for giving a shit.
                </h1>
              </div>,
              "className": "container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1>
                  Get paid for giving a shit.
                </h1>,
                "className": "banner",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Get paid for giving a shit.",
                },
                "ref": null,
                "rendered": "Get paid for giving a shit.",
                "type": "h1",
              },
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
            >
              <h1>
                Get paid for giving a shit.
              </h1>
              <p
                className="text"
              >
                Aute Lorem commodo ipsum dolore laborum enim laborum commodo et ea. Aliquip elit velit aliquip in ex laboris consectetur sit proident esse cillum. Minim duis ad Lorem excepteur consectetur commodo fugiat. Enim consectetur duis ut culpa voluptate labore anim eu dolor deserunt non veniam est. Aute fugiat ea nisi eiusmod anim ex est et exercitation dolore sint in. Labore magna ullamco do dolor sint consectetur. Cillum qui ex do dolor irure magna sint ea dolor nulla ad proident aliqua nostrud.
              </p>
            </div>,
            "className": "text-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Get paid for giving a shit.
                </h1>,
                <p
                  className="text"
                >
                  Aute Lorem commodo ipsum dolore laborum enim laborum commodo et ea. Aliquip elit velit aliquip in ex laboris consectetur sit proident esse cillum. Minim duis ad Lorem excepteur consectetur commodo fugiat. Enim consectetur duis ut culpa voluptate labore anim eu dolor deserunt non veniam est. Aute fugiat ea nisi eiusmod anim ex est et exercitation dolore sint in. Labore magna ullamco do dolor sint consectetur. Cillum qui ex do dolor irure magna sint ea dolor nulla ad proident aliqua nostrud.
                </p>,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Get paid for giving a shit.",
                },
                "ref": null,
                "rendered": "Get paid for giving a shit.",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Aute Lorem commodo ipsum dolore laborum enim laborum commodo et ea. Aliquip elit velit aliquip in ex laboris consectetur sit proident esse cillum. Minim duis ad Lorem excepteur consectetur commodo fugiat. Enim consectetur duis ut culpa voluptate labore anim eu dolor deserunt non veniam est. Aute fugiat ea nisi eiusmod anim ex est et exercitation dolore sint in. Labore magna ullamco do dolor sint consectetur. Cillum qui ex do dolor irure magna sint ea dolor nulla ad proident aliqua nostrud.",
                  "className": "text",
                },
                "ref": null,
                "rendered": "Aute Lorem commodo ipsum dolore laborum enim laborum commodo et ea. Aliquip elit velit aliquip in ex laboris consectetur sit proident esse cillum. Minim duis ad Lorem excepteur consectetur commodo fugiat. Enim consectetur duis ut culpa voluptate labore anim eu dolor deserunt non veniam est. Aute fugiat ea nisi eiusmod anim ex est et exercitation dolore sint in. Labore magna ullamco do dolor sint consectetur. Cillum qui ex do dolor irure magna sint ea dolor nulla ad proident aliqua nostrud.",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
